packages/neuron-wallet/src/services/addresses.ts

```diff
@@ -445,4 +449,40 @@ export default class AddressService {
   public static async deleteByWalletId(walletId: string): Promise<void> {
     await getConnection().createQueryBuilder().delete().from(HdPublicKeyInfo).where({ walletId }).execute()
   }
+
+  public static async getPrivateKeyByAddress({
+    walletID,
+    assetAccountId,
+    password,
+    address,
+  }: {
+    walletID: string
+    assetAccountId?: string
+    password: string
+    address?: string
+  }): Promise<string> {
+    const wallet = WalletService.getInstance().get(walletID)
+    const addresses = await AddressService.getAddressesByWalletId(walletID)
+    let addr = address ? addresses.find(addr => addr.address === address) : addresses[0]
+
+    if (assetAccountId) {
+      const assetAccount = await AssetAccountService.getAccount({ walletID, id: Number(assetAccountId) })
+      if (!assetAccount) {
+        throw new AddressNotFound()
+      }
+      addr = addresses.find(a => a.blake160 === assetAccount.blake160)
+    }
+
+    if (!addr) {
+      throw new AddressNotFound()
+    }
```

Comment:
Maybe we can decode the `address` to a script. Then we can check the address with `blake160`. Then it's unnecessary to transfer `assetAccountId` and `getAccount`.

```
const lock = addressToScript(address)
const addr = addresses.find(a => a.blake160 === lock.args)
```